src/main/java/org/monarchinitiative/biodownload/BioDownloaderBuilder.java

```diff
@@ -0,0 +1,224 @@
+     */
+    public BioDownloaderBuilder proSite() {
+        resources.add(DownloadableResource.PROSITE);
+        return this;
+    }
+
+    /**
+     * Download Hgnc file
+     * @return a builder instance
+     */
+    public BioDownloaderBuilder hgnc() {
+        resources.add(DownloadableResource.HGNC);
+        return this;
+    }
+
+    public BioDownloaderBuilder goGaf() {
+        resources.add(DownloadableResource.GO_GAF);
+        return this;
+    }
+
+    // TODO unzip downloaded zipped files
+    public BioDownloaderBuilder goGafGz() {
+        resources.add(DownloadableResource.GO_GAFGZ);
+        return this;
+    }
+
+    public BioDownloaderBuilder mondoJson() {
+        resources.add(DownloadableResource.MONDO_JSON);
+        return this;
+    }
+
+    public BioDownloaderBuilder mondoOwl() {
+        resources.add(DownloadableResource.MONDO_OWL);
+        return this;
+    }
+
+    public BioDownloaderBuilder mondo() {
+        return mondoOwl().mondoJson();
+    }
+
+    public BioDownloaderBuilder ectoJson() {
+        resources.add(DownloadableResource.ECTO_JSON);
+        return this;
+    }
+
+    public BioDownloaderBuilder ectoOwl() {
+        resources.add(DownloadableResource.ECTO_OWL);
+        return this;
+    }
+
+    public BioDownloaderBuilder ecto() {
+        return ectoOwl().ectoJson();
+    }
+
+    public BioDownloaderBuilder maxoJson() {
+        resources.add(DownloadableResource.MAXO_JSON);
+        return this;
+    }
+
+    public BioDownloaderBuilder maxoOwl() {
+        resources.add(DownloadableResource.MAXO_OWL);
+        return this;
+    }
+
+    public BioDownloaderBuilder maxoObo() {
+        resources.add(DownloadableResource.MAXO_OBO);
+        return this;
+    }
+
+    public BioDownloaderBuilder maxo() {
+        return maxoOwl().maxoObo().maxoJson();
+    }
+
+    public BioDownloaderBuilder hpoJson() {
+        resources.add(DownloadableResource.HP_JSON);
+        return this;
+    }
+
+    public BioDownloaderBuilder hpoObo() {
+        resources.add(DownloadableResource.HP_OBO);
+        return this;
+    }
+
+    public BioDownloaderBuilder hpo() {
+        return hpoObo().hpoJson();
+    }
+
+    public BioDownloaderBuilder hpAnnotation() {
+        resources.add(DownloadableResource.PHENOTYPE_HP);
+        return this;
+    }
+
+    public BioDownloaderBuilder geneInfo() {
+        resources.add(DownloadableResource.GENE_INFO);
+        return this;
+    }
+
+    public BioDownloaderBuilder all() {
+        return go().goGaf().goGafGz().hpo().mondo().ecto().hgnc().maxo().medgene2MIM().proSite().hpAnnotation().geneInfo();
+    }
+
+    /**
+     * Download file from given URL and save it with the given name
+     * @param name Name of file to be downloaded to
+     * @param urlStr URL string of source file
+     * @return a builder instance
+     */
+    public BioDownloaderBuilder custom(String name, String urlStr) {
+        URL url = DownloadableResource.createURL(urlStr);
```

Comment:
This is something we should discuss. On the line below, a `DownloadableResource` with `null` is potentially added into `resources`, which unnecessarily complicates the logic. 
My preference is not having null-containing value objects in the app, unless the null represents an optional attribute. However, I think `DownloadableResource` with `null` is useless, it only complicates the logic. I would prefer `DownloadableResource.url` to be non-nullable.

Reply:
resolved in e89e320